src/components/CouponDetails/helpers.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable import/prefer-default-export */
+import { SINGLE_USE, ONCE_PER_CUSTOMER } from '../../data/constants/coupons';
+import {
+  ACTION_LABELS, ACTION_TYPES, COUPON_FILTER_TYPES, FILTER_OPTIONS,
+} from './constants';
+
+export const getFilterOptions = (usageLimitation) => {
+  const shouldHidePartialRedeem = [SINGLE_USE, ONCE_PER_CUSTOMER].includes(usageLimitation);
+  let options = FILTER_OPTIONS;
+
+  if (shouldHidePartialRedeem) {
+    options = FILTER_OPTIONS.filter(option => option.value !== COUPON_FILTER_TYPES.partiallyRedeemed);
+  }
+
+  return options;
+};
+
+export const getFirstNonDisabledOption = (options) => {
+  const firstNonDisabledOption = options.find(option => !option.disabled);
+
+  if (firstNonDisabledOption) {
+    return firstNonDisabledOption.value;
+  }
+
+  return options[0].value;
+};
+
+export const getBASelectOptions = ({
+  isAssignView,
+  isRedeemedView,
+  hasTableData,
+  couponAvailable,
+  numUnassignedCodes,
+  numSelectedCodes,
+}) => ([{
+  label: ACTION_LABELS.assign,
+  value: ACTION_TYPES.assign,
+  disabled: !isAssignView || isRedeemedView || !hasTableData || !couponAvailable || numUnassignedCodes === 0, // eslint-disable-line max-len
```

Comment:
so many logic gates! 😮

Reply:
This is after I removed one about visibility!